src/naive/date/mod.rs

```diff
@@ -1195,58 +1195,51 @@ impl NaiveDate {
 
     /// Makes a new `NaiveDate` with the year number changed, while keeping the same month and day.
     ///
-    /// This method assumes you want to work on the date as a year-month-day value. Don't use it if
-    /// you want the ordinal to stay the same after changing the year, of if you want the week and
-    /// weekday values to stay the same.
-    ///
     /// # Errors
     ///
-    /// Returns `None` if:
-    /// - The resulting date does not exist (February 29 in a non-leap year).
-    /// - The year is out of range for a `NaiveDate`.
+    /// - Returns [`Error::DoesNotExist`] when the resulting date does not exist.
+    /// - Returns [`Error::OutOfRange`] if `year` is out of range for a `NaiveDate`.
     ///
     /// # Examples
     ///
     /// ```
-    /// use chrono::NaiveDate;
-    ///
-    /// assert_eq!(
-    ///     NaiveDate::from_ymd(2015, 9, 8).unwrap().with_year(2016),
-    ///     Some(NaiveDate::from_ymd(2016, 9, 8).unwrap())
-    /// );
-    /// assert_eq!(
-    ///     NaiveDate::from_ymd(2015, 9, 8).unwrap().with_year(-308),
-    ///     Some(NaiveDate::from_ymd(-308, 9, 8).unwrap())
-    /// );
+    /// # use chrono::{NaiveDate, Error};
+    /// assert_eq!(NaiveDate::from_ymd(2015, 9, 8)?.with_year(2016),
+    ///            NaiveDate::from_ymd(2016, 9, 8));
```

Comment:
This doesn't follow the rustfmt style anymore #1451.
I run it with `cargo fmt -- --config format_code_in_doc_comments=true` and discard any stray changes.

Reply:
Thanks, I was getting crazy with the comments not formatting automatically.

Reply:
:laughing: With all the changes the renames and `?`s we're  introducing I am glad we have this. Needs nightly though and has its quicks. (And I have a couple of compiles ICEs that I should report...)